vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc

```diff
@@ -142,20 +114,54 @@ void UpholdWallet::OnGetUser(const type::Result result,
     return callback(type::Result::LEDGER_ERROR);
   }
 
-  if (user.status != UserStatus::OK) {
+  ledger_->uphold()->GetCapabilities(std::bind(
+      &UpholdWallet::OnGetCapabilities, this, _1, _2, std::move(callback)));
+}
+
+void UpholdWallet::OnGetCapabilities(type::Result result,
+                                     std::map<std::string, bool> capabilities,
+                                     ledger::ResultCallback callback) const {
+  auto uphold_wallet = ledger_->uphold()->GetWallet();
+  if (!uphold_wallet) {
+    BLOG(0, "Uphold wallet is null!");
+    return callback(type::Result::LEDGER_ERROR);
+  }
+
+  if (uphold_wallet->status != type::WalletStatus::PENDING &&
+      uphold_wallet->status != type::WalletStatus::VERIFIED) {
+    return callback(type::Result::LEDGER_OK);
+  }
+
+  CheckWalletState(uphold_wallet.get());
+
+  if (result == type::Result::EXPIRED_TOKEN) {
+    BLOG(0, "Access token expired!");
     // Entering NOT_CONNECTED or DISCONNECTED_VERIFIED.
-    ledger_->uphold()->DisconnectWallet("");
+    ledger_->uphold()->DisconnectWallet(
+        ledger::notifications::kWalletDisconnected);
+    return callback(type::Result::EXPIRED_TOKEN);
+  }
+
+  if (result != type::Result::LEDGER_OK || capabilities.empty() ||
+      !capabilities.count("receives") || !capabilities.count("sends")) {
```

Comment:
By using string keys here, it seems that this class is taking on some of the "parsing" responsibilities associated with the endpoint code.

Another option might be to have the `GetCapabilities` endpoint class return a structure instead, that contains the information that callers care about (maybe something with `can_send` and `can_receive` bools).

Reply:
@zenparsing Yeah, good point — amended in https://github.com/brave/brave-core/pull/13474/commits/4eba72eaa123b8882409d9d7b4331ab9182512b0.